src/controller/ExamplePersistentStorage.cpp

```diff
@@ -55,36 +55,37 @@ const char * GetUsedDirectory(const char * directory)
         dir = "/tmp";
     }
 
-    return dir;
+    return std::string{ dir };
```

Comment:
nit: I find this "create a variable and keep updating it" somewhat an annoying style. Maybe consider:

```suggestion
    VerifyOrReturnValue(directory.empty(), directory);
    
    if (const char *dir = getenv("TMPDIR"); dir != nullptr) {
        return dir;
    }
    
    return "/tmp";
```

Reply:
Done